File: java/mandelbrot/Simple.java
package mandelbrot;

public final class Simple {
    public static final int SIZE = 40;

    public static void index() {
        for (var y = -39; y < 39; y++) {
            System.out.println();

            for (var x = -39; x < 39; x++) {
                int i = mandelbrot(
                    x / 40.0,
                    y / 40.0
                );

                if (i == 0) {
                    System.out.print("*");
                } else {
                    System.out.print(" ");
                }
            }
        }
    }

    public static int mandelbrot(double x, double y) {
        double cr = y - 0.5;
        double ci = x;
        double zi = 0.0;
        double zr = 0.0;
        int i = 0;

        while (true) {
            i++;

            double temp = zr * zi;

            double zr2 = zr * zr;
            double zi2 = zi * zi;

            zr = zr2 - zi2 + cr;
            zi = temp + temp + ci;

            if (zi2 + zr2 > 16) {
                return i;
            }

            if (i > 5000) {
                return 0;
            }
        }
    }

    public static void main(String[] args) {
        index();
    }
}

File: lua/mandelbrot/Simple.lua
#!/usr/bin/env lua

function index()
    for y = -39, 38 do
        io.write("\n")
        
        for x = -39, 38 do
            local i = mandelbrot(
                x / 40.0,
                y / 40.0
            )

            if i == 0 then
                io.write("*")
            else
                io.write(" ")
            end
        end
    end

    io.write("\n")
end

function mandelbrot(x, y)
    local cr = y - 0.5
    local ci = x
    local zi = 0.0
    local zr = 0.0
    local i = 0

    while true do
        i = i + 1
        
        local temp = zr * zi
        
        local zr2 = zr * zr
        local zi2 = zi * zi
        
        zr = zr2 - zi2 + cr
        zi = temp + temp + ci
        
        if zi2 + zr2 > 16 then
            return i
        end
        
        if i > 5000 then
            return 0
        end
    end
end

local function main()
    local startTime = os.clock()
    
    index()
    
    local endTime = os.clock()
    local duration = (endTime - startTime) * 1000
    
    print(string.format("Execution time: %.0fms", duration))
end

main()
